client/allocrunner/csi_hook.go

```diff
@@ -85,11 +85,15 @@ func (c *csiHook) Prerun() error {
 	mounts := make(map[string]*csimanager.MountInfo, len(volumes))
 	for alias, pair := range volumes {
 
-		// We use this context only to attach hclog to the gRPC
-		// context. The lifetime is the lifetime of the gRPC stream,
-		// not specific RPC timeouts, but we manage the stream
-		// lifetime via Close in the pluginmanager.
-		mounter, err := c.csimanager.MounterForPlugin(c.shutdownCtx, pair.volume.PluginID)
+		// make sure the plugin is ready or becomes so quickly.
```

Comment:
I think this comment is misleading, because we're actually waiting up to 24hr? 

Having the really long timeout on the unpublish workflow makes sense because we need to be able to recover the already-mounted volume from that state. But on the publish workflow we probably want to abandon the attempt within a short-ish amount of time so that we're not delaying a reschedule too much in the case where we've placed a new alloc and it's never going to work. 

Maybe... 5min to allow for a _very_ busy restoring client?

(Ideally we'd be able to distinguish between a "this is the first time" vs a "this is a restore", but we can do that and/or make the timeout tunable later.)

Reply:
Up to 24 hours is the last catch-all timeout [on the registry end](https://github.com/hashicorp/nomad/pull/16809/files#diff-533c7964b7d08aa5521e171e5200f35b4618cb7cd53348491b6aafc82d295611R317), just in case callers don't set their own timeouts like they should.  The csiManager's WaitForPlugin() [adds a timeout of 1 Minute](https://github.com/hashicorp/nomad/pull/16809/files#diff-da66b90315db1e7fce5ec609ab6eb1656a48c452e807527caf0416552770d040R81) because it has (or can have) more specific domain knowledge of its plugins' behavior.

I went back and forth on where to put which contexts with which timeout durations, and these values worked okay in my little sandbox.  Should I bump csiManager's up to 5min?  Or move them around some otherlyways?

Reply:
Ah, ok, I missed the 1 min on `csimanager.WaitForPlugin`. If we end up refactoring the postrun hook to use `csimanager.WaitForPlugin` too, we might find we want to lift that timeout configuration into the caller (the prerun or postrun hook), but this should be fine as-is for now.